### DISCUSSION TITLE
Splitstage should not be hardcoded and instead should be read from the skin.ini file.

### DISCUSSION THREAD

**Discussion by Hydria** [ppy/osu#23916] [Ideas]
Currently at the moment, any keymode above 10K is forced to use Split Stage, whilst any mode below 10K should have to use Dual Stages mod to have split stages (which also doesn't function at the moment by the looks of it).

You can see this enforced in `ManiaRuleset.cs` under `MAX_STAGE_KEYS` being set to 10, meaning that `AvailableVariants` sets the mode to Single/Dual depending on the keymode alone, and not checking against the skin.

This would also require edits to `SingleStageVarientGenerator.cs` file to support the higher keymode key list.

The main reason this discussion has been brought up is potential talks towards allowing 12K maps into the mania ranked section, and players more experienced in playing 12 Keys have brought up that they want to play with a single stage varient as that is what they're used to on stable.

**Comment by AkiraMorilas** [comment]
Related issue: https://github.com/ppy/osu/issues/23620

Also, considering that very few players use split stages at all, and even then usually only in 14K2S, unsplitting them even in higher keycounts, wouldn't it be fine to just default to a single stage for everything, only splitting if Dual Stages mod is selected or skin has `SplitStages:` set to 1 for that keycount?
